scapy/sendrecv.py

```diff
@@ -462,9 +462,22 @@ def sr(x, promisc=None, filter=None, iface=None, nofilter=0, *args, **kargs):
     return result
 
 
+def _interface_selection(iface, packet):
+    """
+    Select the network interface according to the layer 3 destination
+    """
+
+    if iface is None and getattr(packet, "route"):
+        iff, addr, gw = packet.route()
```

Comment:
You could probably discard the unused values

Reply:
Done. Thanks!